packages/grid/x-data-grid/src/utils/utils.ts

```diff
@@ -38,3 +34,117 @@ export function escapeRegExp(value: string): string {
  */
 export const clamp = (value: number, min: number, max: number) =>
   Math.max(min, Math.min(max, value));
+
+/**
+ * Based on `fast-deep-equal`
+ * We only type the public interface to avoid dozens of `as` in the function.
+ */
+export function isDeepEqual<T>(actual: any, expected: T): actual is T;
+
+export function isDeepEqual(a: any, b: any) {
```

Comment:
I kept all the features of `fast-deep-equal` except the React object comparison
But it is over-powered for the usage we have right now